frontend/src/containers/GameChangerDetailsPage.js

```diff
@@ -262,6 +268,43 @@ const GameChangerDetailsPage = (props) => {
 		return data;
 	};
 
+	const getHeadData = async (name, cloneName) => {
+		const resp = await gameChangerAPI.callGraphFunction({
+			functionName: 'getHeadDataDetailsPage',
+			cloneName: cloneName,
+			options: {
+				headName: name,
+			},
+		});
+		if (resp.data.headData.head) {
+			return resp.data.headData.head;
+		} else {
+			return '';
+		}
+	};
+
+	const getTypeData = async (name, cloneName) => {
+		let types = '';
+		const resp = await gameChangerAPI.callGraphFunction({
+			functionName: 'getTypeDataDetailsPage',
+			cloneName: cloneName,
+			options: {
+				typeName: name,
+			},
+		});
+		if (resp.data.typeData.nodes && resp.data.typeData.nodes.length > 0) {
```

Comment:
Whenever parsing arrays, array functions should be used in place of raw for loops https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join

Optional chaining (`?.` syntax) allows for null/undefined checking in-line. The entire expression will return undefined if any of the checked properties are nullish https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

combine this with another undefined check using logical OR and you get this whole section in 1 short line of code 🙂

`const types = resp?.data?.typeData?.nodes?.join(', ') || '';`